x/crosschain/keeper/keeper_cross_chain_tx_prove_outbound_tx.go

```diff
@@ -0,0 +1,195 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observerTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// Casts a vote on an outbound transaction observed on a connected chain (after
+// it has been broadcasted to and finalized on a connected chain). If this is
+// the first vote, a new ballot is created. When a threshold of votes is
+// reached, the ballot is finalized. When a ballot is finalized, the outbound
+// transaction is processed.
+//
+// If the observation is successful, the difference between zeta burned
+// and minted is minted by the bank module and deposited into the module
+// account.
+//
+// If the observation is unsuccessful, the logic depends on the previous
+// status.
+//
+// If the previous status was `PendingOutbound`, a new revert transaction is
+// created. To cover the revert transaction fee, the required amount of tokens
+// submitted with the CCTX are swapped using a Uniswap V2 contract instance on
+// ZetaChain for the ZRC20 of the gas token of the receiver chain. The ZRC20
+// tokens are then
+// burned. The nonce is updated. If everything is successful, the CCTX status is
+// changed to `PendingRevert`.
+//
+// If the previous status was `PendingRevert`, the CCTX is aborted.
+//
+// ```mermaid
+// stateDiagram-v2
+//
+//	state observation <<choice>>
+//	state success_old_status <<choice>>
+//	state fail_old_status <<choice>>
+//	PendingOutbound --> observation: Finalize outbound
+//	observation --> success_old_status: Observation succeeded
+//	success_old_status --> Reverted: Old status is PendingRevert
+//	success_old_status --> OutboundMined: Old status is PendingOutbound
+//	observation --> fail_old_status: Observation failed
+//	fail_old_status --> PendingRevert: Old status is PendingOutbound
+//	fail_old_status --> Aborted: Old status is PendingRevert
+//	PendingOutbound --> Aborted: Finalize outbound error
+//
+// ```
+//
+// Only observer validators are authorized to broadcast this message.
+func (k msgServer) ProveOutboundTx(goCtx context.Context, msg *types.MsgProveOutboundTx) (*types.MsgProveOutboundTxResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	// verify proof
+	blockHash := eth.HexToHash(msg.BlockHash)
+	res, found := k.zetaObserverKeeper.GetBlockHeader(ctx, blockHash.Bytes())
+	if !found {
+		return nil, sdkerrors.Wrapf(observerTypes.ErrBlockHeaderNotFound, "Block header not found: %s", msg.BlockHash)
+	}
+	chainID := res.ChainId
+
+	var header ethtypes.Header
+	err := rlp.DecodeBytes(res.Header, &header)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(observerTypes.ErrBlockHeaderNotFound, "cannot decode block header: %s", msg.BlockHash)
+	}
+	var txx ethtypes.Transaction
```

Comment:
Check the CCTX status using nonce before  ? We dont need to verify any proofs if the CCTX is already processed ?